### PR TITLE
rubocop: lib/jekyll/regenerator.rb

### PR DESCRIPTION
**Pull request by ayastreb**
#4885

### CODE REVIEW

File: lib/jekyll/regenerator.rb [hunk 1]
```diff
@@ -20,18 +20,14 @@ def initialize(site)
...
-            document.path, document.destination(@site.dest)
-          )
+        regenerate_document?(document)
       else
-        source_path = document.respond_to?(:path)        ? document.path                    : nil
-        dest_path   = document.respond_to?(:destination) ? document.destination(@site.dest) : nil
+        source_path = document.respond_to?(:path) ? document.path : nil
+        dest_path = if document.respond_to?(:destination)
+                      document.destination(@site.dest)
+                    end
```
Comment (parkr) [inline lib/jekyll/regenerator.rb]: What was wrong with the previous two lines? Just too long?

It seems like your solution for `source_path` and for `dest_path` are two ways of doing the same thing.

File: lib/jekyll/regenerator.rb [hunk 2]
```diff
@@ -173,5 +153,42 @@ def read_metadata
...
+            return cache[dependency] = cache[path] = true
+          end
+        end
+        if File.exist?(path) && data["mtime"].eql?(File.mtime(path))
+          return cache[path] = false
+        end
+      end
+
+      # Path does not exist in metadata, add it
+      add(path)
```
Comment (parkr) [inline lib/jekyll/regenerator.rb]: `check_path_exists` should not modify anything.

File: lib/jekyll/regenerator.rb [hunk 3]
```diff
@@ -94,23 +90,7 @@ def modified?(path)
...
-        if File.exist?(path) && data["mtime"].eql?(File.mtime(path))
-          return cache[path] = false
-        else
-          return add(path)
-        end
-      end
-
-      # Path does not exist in metadata, add it
-      return add(path)
+      check_path_exists(path)
```
Comment (parkr) [inline lib/jekyll/regenerator.rb]: The comment on the old line 97 only referenced the old line 98. 99 thru 110 determined if the file was modified, so it has nothing to do with the comment. This is three operations:

1. If we have seen this file before and one of its dependencies have been modified, set the regeneration bit for both the dependency and the file to `true`
2. If we have seen this file before and it has not been modified, set the regeneration bit to `false`. If it has been modified, set it to `true`.
3. If we have not seen this file before, add it to the metadata and regenerate it.

File: lib/jekyll/regenerator.rb (hunk 1 above)
Comment (ayastreb) [inline lib/jekyll/regenerator.rb]: Yes, `dest_path` line was 98 chars long.
Is there another way to make it shorter?)

**Comment by ayastreb** [comment]
Hi @parkr,
I've reworked `modified?` method a bit, hope now it's more clear 😄

**Comment by parkr** [comment]
This looks GREAT! Well done, thank you ❤️ 

@jekyllbot: merge +dev

**Pull request merged**